lib/tzinfo/timezone.rb

```diff
@@ -418,6 +418,21 @@ def period_for_local(local, dst = Timezone.default_dst)
       end
     end
 
+    def period_for(tm, dst = Timezone.default_dst)
+      tm = tm.to_orig if tm.is_a?(TimeOrDateTime)
+
+      utc = case tm
+      when Time
+        tm.utc
```

Comment:
`Time#utc` modifies the receiver (which should be avoided) instead of creating a new instance.